### PR TITLE
fixed unused code, error prone and code style 

### PR DESCRIPTION
**Pull request by zsdonghao**
- fixed all unused code
- fixed all error prone
- fixed most of the code style

### CODE REVIEW

File: example/tutorial_vgg16.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def conv_layers(net_in):
 
 
 def conv_layers_simple_api(net_in):
-    with tf.name_scope('preprocess') as scope:
+    with tf.name_scope('preprocess'):  # as scope:
```
Comment (luomai) [inline example/tutorial_vgg16.py]: remove this comment as users may compensate as a description.

File: tensorlayer/layers/core.py [hunk 2]
```diff
@@ -702,7 +707,7 @@ def __init__(
         self.inputs = inputs
         logging.info("EmbeddingInputlayer %s: (%d, %d)" % (self.name, vocabulary_size, embedding_size))
 
-        with tf.variable_scope(name) as vs:
+        with tf.variable_scope(name):  # as vs:
```
Comment (luomai) [inline tensorlayer/layers/core.py]: remove comment

File: tensorlayer/layers/extend.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def __init__(
         self.inputs = layer.outputs
 
         logging.info("TileLayer  %s: multiples:%s" % (self.name, multiples))
-        with tf.variable_scope(name) as vs:
+        with tf.variable_scope(name):  # as vs:
```
Comment (luomai) [inline tensorlayer/layers/extend.py]: remove comment

File: tensorlayer/layers/merge.py [hunk 4]
```diff
@@ -112,8 +112,10 @@ def __init__(
         self.outputs = layers[0].outputs
         # logging.info(self.outputs._shape, type(self.outputs._shape))
         for l in layers[1:]:
-            assert str(self.outputs.get_shape()) == str(
-                l.outputs.get_shape()), "Hint: the input shapes should be the same. %s != %s" % (self.outputs.get_shape(), str(l.outputs.get_shape()))
+            # assert str(self.outputs.get_shape()) == str(
```
Comment (luomai) [inline tensorlayer/layers/merge.py]: remove comment

File: tensorlayer/layers/recurrent.py [hunk 5]
```diff
@@ -1382,7 +1383,7 @@ def __init__(
             if return_last:
                 # [batch_size, 2 * n_hidden]
                 raise Exception("Do not support return_last at the moment")
```
Comment (luomai) [inline tensorlayer/layers/recurrent.py]: shall we actually remove this parameter from the interface?

File: tensorlayer/layers/stack.py [hunk 6]
```diff
@@ -68,7 +68,7 @@ def unstack_layer(layer, num=None, axis=0, name='unstack'):
 
     """
     inputs = layer.outputs
-    with tf.variable_scope(name) as vs:
+    with tf.variable_scope(name):  # as vs:
```
Comment (luomai) [inline tensorlayer/layers/stack.py]: remove comment

File: tensorlayer/layers/super_resolution.py [hunk 7]
```diff
@@ -94,7 +94,7 @@ def _PS(X, r, n_out_channels):
 
     net_new = Layer(inputs, name=whole_name)
     # with tf.name_scope(name):
-    with tf.variable_scope(name) as vs:
+    with tf.variable_scope(name):  # as vs:
```
Comment (luomai) [inline tensorlayer/layers/super_resolution.py]: remove comment

File: tensorlayer/prepro.py [hunk 8]
```diff
@@ -2195,9 +2205,12 @@ def parse_darknet_ann_list_to_cls_box(annotations):
     """
     class_list = []
     bbox_list = []
-    for i in range(len(annotations)):
-        class_list.append(annotations[i][0])
-        bbox_list.append(annotations[i][1:])
+    # for i in range(len(annotations)):
```
Comment (luomai) [inline tensorlayer/prepro.py]: remove comments

File: tensorlayer/utils.py [hunk 9]
```diff
@@ -542,8 +553,7 @@ def exit_tensorflow(sess=None, port=6006):
         logging.info('OS X: %s' % text)
         subprocess.Popen("lsof -i tcp:" + str(port) + "  | grep -v PID | awk '{print $2}' | xargs kill", shell=True)  # kill tensorboard
     elif _platform == "win32":
-        logging.info(text2 + "Windows")
-        # TODO
+        raise NotImplementedError()
```
Comment (luomai) [inline tensorlayer/utils.py]: better to add a string telling "this function is not supported on the Windows platform"

File: example/tutorial_vgg16.py (hunk 1 above)
Comment (lgarithm) [inline example/tutorial_vgg16.py]: remove `# as scope`

File: tensorlayer/utils.py [hunk 10]
```diff
@@ -566,16 +576,14 @@ def open_tensorboard(log_dir='/tmp/tensorflow', port=6006):
...
     elif _platform == "darwin":
         logging.info('OS X: %s' % text)
         subprocess.Popen(
             sys.prefix + " | python -m tensorflow.tensorboard --logdir=" + log_dir + " --port=" + str(port),
             shell=True)  # open tensorboard in localhost:6006/ or whatever port you chose
     elif _platform == "win32":
-        logging.info('Windows%s' % text2)
-        # TODO
+        raise NotImplementedError()
```
Comment (luomai) [inline tensorlayer/utils.py]: the same as above

File: tensorlayer/layers/core.py [hunk 11]
```diff
@@ -289,11 +289,14 @@ def list_remove_repeat(x):
 
     """
     y = []
-    [y.append(i) for i in x if not i in y]
+    # [y.append(i) for i in x if not i in y]
```
Comment (luomai) [inline tensorlayer/layers/core.py]: remove the comment

File: tensorlayer/layers/recurrent.py [hunk 12]
```diff
@@ -1325,7 +1325,8 @@ def __init__(
                                       input_keep_prob=in_keep_prob,
                                       output_keep_prob=out_keep_prob if is_last else 1.0)  # out_keep_prob)
             else:
-                cell_creator = lambda: rnn_creator()
+                # cell_creator = lambda: rnn_creator()
```
Comment (luomai) [inline tensorlayer/layers/recurrent.py]: remove comment

File: tensorlayer/nlp.py [hunk 13]
```diff
@@ -67,8 +67,13 @@ def generate_skip_gram_batch(data, batch_size, num_skips, skip_window, data_inde
     # global data_index   # you can put data_index outside the function, then
     #       modify the global data_index in the function without return it.
     # note: without using yield, this code use data_index to instead.
-    assert batch_size % num_skips == 0
-    assert num_skips <= 2 * skip_window
+
+    # assert batch_size % num_skips == 0
+    if batch_size % num_skips != 0:
+        raise Exception("batch_size should be able to be divided by num_skips.")
+    # assert num_skips <= 2 * skip_window
```
Comment (luomai) [inline tensorlayer/nlp.py]: remove

File: tensorlayer/nlp.py [hunk 14]
```diff
@@ -67,8 +67,13 @@ def generate_skip_gram_batch(data, batch_size, num_skips, skip_window, data_inde
     # global data_index   # you can put data_index outside the function, then
     #       modify the global data_index in the function without return it.
     # note: without using yield, this code use data_index to instead.
-    assert batch_size % num_skips == 0
-    assert num_skips <= 2 * skip_window
+
+    # assert batch_size % num_skips == 0
```
Comment (luomai) [inline tensorlayer/nlp.py]: remove

File: tensorlayer/nlp.py [hunk 15]
```diff
@@ -665,12 +674,15 @@ def build_words_dataset(words=[], vocabulary_size=50000, printable=True, unk_key
     if printable:
         logging.info('Real vocabulary size    %d' % len(collections.Counter(words).keys()))
         logging.info('Limited vocabulary size {}'.format(vocabulary_size))
-    assert len(collections.Counter(words).keys()) >= vocabulary_size, \
-        "the limited vocabulary_size must be less than or equal to the read vocabulary_size"
+    # assert len(collections.Counter(words).keys()) >= vocabulary_size, \
```
Comment (luomai) [inline tensorlayer/nlp.py]: remove

**Pull request merged**